n2survey/lime/survey.py

```diff
@@ -458,19 +462,89 @@ def plot(
 
         question_type = self.get_question_type(question)
+        # get plot title
+        if plot_title:
+            if type(plot_title) != str:
+                plot_title = self.get_label(question)
 
         if question_type == "single-choice":
             counts_df = self.count(question, labels=True)
 
             if "title" not in non_theme_kwargs:
                 non_theme_kwargs.update({"title": counts_df.columns[0]})
 
-            fig, ax = single_choice_bar_plot(
-                x=counts_df.index.values,
-                y=pd.Series(counts_df.iloc[:, 0], name="Number of Responses"),
-                theme=theme,
-                **non_theme_kwargs,
-            )
+            if compare_with:
+                DF = []
+                if self.get_question_type(compare_with) == "single-choice":
+
+                    # create Dataframe from both questions
+                    DF.append(
+                        pd.concat(
+                            [
+                                self.get_responses(
+                                    question, labels=True, drop_other=True
+                                ),
+                                self.get_responses(
+                                    compare_with, labels=True, drop_other=True
+                                ),
+                            ],
+                            axis=1,
+                        ).values
+                    )
+
+                else:
+                    raise NotImplementedError(
+                        """
+                         only single-choice to single-choice comparison
+                         implemented at the moment.
+                         """
+                    )
+                if add_questions:
+                    for entry in add_questions:
+                        if self.get_question_type(compare_with) == "single-choice":
+                            DF.append(
+                                pd.concat(
+                                    [
+                                        self.get_responses(
+                                            entry, labels=True, drop_other=True
+                                        ),
+                                        self.get_responses(
+                                            compare_with, labels=True, drop_other=True
+                                        ),
+                                    ],
+                                    axis=1,
+                                ).values
+                            )
+                        else:
+                            raise NotImplementedError(
+                                """
+                            only single-choice to single-choice comparison
+                            implemented at the moment.
+                            """
+                            )
+                if totalbar:
+                    totalbar = np.unique(
```

Comment:
I think I fixed it partially, it is now still boolean for the End user, so they have to just type plot(... totalbar=True ...)' which is easier for them, so they don't have to think about calculating it but the plotting function itself ('simple_comparison_plot') now needs an np.array or none insteaad of boolean, so I don't have to add more parameters, because there are already a lot of them... maybe later on I can add the calculation of the totalbar to simple_comparison but for now this should be better.